go-controller/cmd/ovn-k8s-cni-overlay/app/helper_linux.go

```diff
@@ -103,8 +103,69 @@ func setupInterface(netns ns.NetNS, containerID, ifName, macAddress, ipAddress,
 	return hostIface, contIface, nil
 }
 
+func clearPodBandwidth(sandboxID string) error {
+	// interfaces will have the same name as ports
+	portList, err := ovsFind("interface", "name", "external-ids:sandbox="+sandboxID)
+	if err != nil {
+		return err
+	}
+
+	// Clear the QoS for any ports of this sandbox
+	for _, port := range portList {
+		if err = ovsClear("port", port, "qos"); err != nil {
+			return err
+		}
+	}
+
+	// Now that the QoS is unused remove it
+	qosList, err := ovsFind("qos", "_uuid", "external-ids:sandbox="+sandboxID)
+	if err != nil {
+		return err
+	}
+	for _, qos := range qosList {
+		if err := ovsDestroy("qos", qos); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func setPodBandwidth(sandboxID, hostVeth string, ingressBPS, egressBPS int64) error {
+	// note pod ingress == OVS egress and vice versa
+
+	l, err := netlink.LinkByName(hostVeth)
+	if err != nil {
+		return fmt.Errorf("failed to find host veth interface %s: %v", hostVeth, err)
+	}
+	err = netlink.LinkSetTxQLen(l, 1000)
+	if err != nil {
+		return fmt.Errorf("failed to set host veth txqlen: %v", err)
+	}
+
+	if ingressBPS > 0 {
+		qos, err := ovsCreate("qos", "type=linux-htb", fmt.Sprintf("other-config:max-rate=%d", ingressBPS), "external-ids=sandbox="+sandboxID)
+		if err != nil {
+			return err
+		}
+		err = ovsSet("port", hostVeth, fmt.Sprintf("qos=%s", qos))
+		if err != nil {
+			return err
+		}
+	}
+	if egressBPS > 0 {
+		// ingress_policing_rate is in Kbps
+		err := ovsSet("interface", hostVeth, fmt.Sprintf("ingress_policing_rate=%d", egressBPS/1024))
```

Comment:
I'm wondering if it shouldn't be `/1000`. [OVS docs for QoS](https://github.com/openvswitch/ovs/blob/master/Documentation/faq/qos.rst) give examples using 1000's as multiplier (i.e. kilo- not kibi- bps).

Reply:
@jsitnicki good question, I'm happy to change it

Reply:
@jsitnicki fixed.